src/main/webapp/app/entities/oauth-client/oauth-client-dialog.component.html

```diff
@@ -0,0 +1,75 @@
+            <input *ngIf="newClient" type="text" class="form-control" id="id" name="id" [(ngModel)]="client.clientId" required />
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.clientSecret">Client Secret</label>
+            <div class="input-group">
+                <input type="text" class="form-control" id="secret" name="clientSecret" [(ngModel)]="client.clientSecret" [disabled]="protectedClient" [required]="newClient" />
+                <span class="input-group-btn"><button class="btn btn-secondary" type="button" (click)="generateRandomSecret()"><span class="fa fa-magic">&nbsp;</span>Randomize</button></span>
+            </div>
+            <small id="scopeHelp" class="form-text text-muted" *ngIf="!newClient">If this field is left empty the secret will not be modified.</small>
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.scope">Scope</label>
+            <input type="text" class="form-control" id="scope" name="scope" [(ngModel)]="scopeList" [disabled]="protectedClient" required />
+            <small id="scopeHelp" class="form-text text-muted">Comma seperated list of scopes. Scopes should have the following structure: ENTITY.OPERATION. See the <a href="https://github.com/RADAR-CNS/ManagementPortal/tree/master/radar-auth">radar-auth library documentation</a> for more information.</small>
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.resourceIds">Resources</label>
+            <input type="text" class="form-control" id="resourceIds" name="resourceIds" [(ngModel)]="resourcesList" [disabled]="protectedClient" required />
+            <small id="scopeHelp" class="form-text text-muted">Comma seperated list of allowed resource IDs.</small>
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.authorizedGrantTypes">Grant Types</label>
+            <div class="form-check" *ngFor="let grant of grantTypes">
+                <label class="form-check-label">
+                    <input type="checkbox" class="form-check-input" name="grantTypes" [checked]="grant.authorized" (change)="grant.authorized = !grant.authorized" [disabled]="protectedClient" /> {{grant.name}}
+                </label>
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.autoApprove">Auto Approve</label>
+            <input type="text" class="form-control" id="autoApprove" name="autoApprove" [(ngModel)]="autoApproveScopeList" [disabled]="protectedClient" />
```

Comment:
I see that the server doesn't accept if we try to give a scope that is not provided earlier. However, there is no message or feedback given to user when they try to do that.  I think at least it is a good to let the user choose from the scopes that are already provided and it should be easy to implement since we have two way binding.